src/main/java/emissary/place/ComparisonPlace.java

```diff
@@ -0,0 +1,142 @@
+import emissary.util.PlaceComparisonHelper;
+
+import org.apache.commons.lang3.Validate;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This place takes two other places, givens them exactly the same IBDO and compares the output.
+ * <p>
+ * The configuration file must contain the five properties defined by the five constants in this class.
+ */
+public class ComparisonPlace extends ServiceProviderPlace {
+    /**
+     * The full pathname for the class for the first class.
+     */
+    public static final String PLACE_A_CLASSNAME = "PLACE_A_CLASSNAME";
+    /**
+     * The full pathname for the configuration for the first class.
+     */
+    public static final String PLACE_A_CONFIGNAME = "PLACE_A_CONFIGNAME";
+    /**
+     * The full pathname for the class for the second class.
+     */
+    public static final String PLACE_B_CLASSNAME = "PLACE_B_CLASSNAME";
+    /**
+     * The full pathname for the configuration for the second class.
+     */
+    public static final String PLACE_B_CONFIGNAME = "PLACE_B_CONFIGNAME";
+    /**
+     * An identifier to be added to the log message.
+     */
+    public static final String LOGGING_IDENTIFIER = "LOGGING_IDENTIFIER";
+
+    private static final String NOT_NULL_STRING = " == null not allowed!";
+    private static final String PROCESS_PROCESSHD_MSG = "Mixed process/processHeavyDuty not allowed!";
+
+    private final ServiceProviderPlace placeA;
+    private final ServiceProviderPlace placeB;
+    private final String loggingIdentifier;
+
+    public ComparisonPlace(String configFile, String theDir, String thePlaceLocation) throws IOException {
+        super(configFile, theDir, thePlaceLocation);
+
+        final String placeAClassName = configG.findStringEntry(PLACE_A_CLASSNAME);
+        final String placeAConfigName = configG.findStringEntry(PLACE_A_CONFIGNAME);
+        final String placeBClassName = configG.findStringEntry(PLACE_B_CLASSNAME);
+        final String placeBConfigName = configG.findStringEntry(PLACE_B_CONFIGNAME);
+
+        loggingIdentifier = configG.findStringEntry(LOGGING_IDENTIFIER);
+
+        Validate.notNull(placeAClassName, PLACE_A_CLASSNAME + NOT_NULL_STRING);
+        Validate.notNull(placeAConfigName, PLACE_A_CONFIGNAME + NOT_NULL_STRING);
+        Validate.notNull(placeBClassName, PLACE_B_CLASSNAME + NOT_NULL_STRING);
+        Validate.notNull(placeBConfigName, PLACE_B_CONFIGNAME + NOT_NULL_STRING);
+        Validate.notNull(loggingIdentifier, LOGGING_IDENTIFIER + NOT_NULL_STRING);
+
+        placeA = createPlace(placeAClassName, placeAConfigName);
+        placeB = createPlace(placeBClassName, placeBConfigName);
+
+        Validate.isTrue(placeA.processMethodImplemented == placeB.processMethodImplemented, PROCESS_PROCESSHD_MSG);
+    }
+
+    @Override
+    public List<IBaseDataObject> processHeavyDuty(IBaseDataObject ibdoA) throws ResourceException {
```

Comment:
We need a comment either here or within the class Javadoc comment clearly detailing the side effects of having this place configured. 
- How does this place impact the IBDO and any children? 
- How should this place be registered with the routing/directory framework? 
  - What about "PlaceA" and "PlaceB"? 
  - Do any service proxies need to be modified?